decoder/decoder.go

```diff
@@ -42,21 +65,33 @@ func readRequestJSONData(maxSize int64) Reader {
 			return nil, errors.New("no content")
 		}
 
+		cLen := req.ContentLength
+		knownCLen := cLen > -1
 		switch req.Header.Get("Content-Encoding") {
 		case "deflate":
+			if knownCLen {
+				deflateLengthAcc.Add(cLen)
+			}
 			var err error
 			reader, err = zlib.NewReader(reader)
 			if err != nil {
 				return nil, err
 			}
 
 		case "gzip":
+			if knownCLen {
+				gzipLengthAcc.Add(cLen)
+			}
 			var err error
 			reader, err = gzip.NewReader(reader)
 			if err != nil {
 				return nil, err
 			}
 		}
+		if knownCLen {
+			uncompressedLengthAcc.Add(cLen)
+		}
+		io.Pipe()
```

Comment:
Is this a leftover as you don't use the returned `PipeReader` and `PipeWriter`

Reply:
yes, is a leftover